src/utils/thread_helper.cpp

```diff
@@ -1343,25 +1379,85 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
                 CopyNat64TrafficCounters(otMapping.mCounters.mIcmp, nat64MappingCounters->mutable_icmp());
 
                 {
-                    uint8_t ipAddrShaInput[OT_IP6_ADDRESS_SIZE + kNat64SourceAddressHashSaltLength];
+                    uint8_t ipAddrShaInput[OT_IP6_ADDRESS_SIZE + kNat64PdCommonHashSaltLength];
                     memcpy(ipAddrShaInput, otMapping.mIp6.mFields.m8, sizeof(otMapping.mIp6.mFields.m8));
-                    memcpy(&ipAddrShaInput[sizeof(otMapping.mIp6.mFields.m8)], mNat64Ipv6AddressSalt,
-                           sizeof(mNat64Ipv6AddressSalt));
+                    memcpy(&ipAddrShaInput[sizeof(otMapping.mIp6.mFields.m8)], mNat64PdCommonSalt,
+                           sizeof(mNat64PdCommonSalt));
 
                     sha256.Start();
                     sha256.Update(ipAddrShaInput, sizeof(ipAddrShaInput));
                     sha256.Finish(hash);
 
                     nat64Mapping->mutable_hashed_ipv6_address()->append(reinterpret_cast<const char *>(hash.GetBytes()),
-                                                                        sizeof(hash.GetBytes()));
+                                                                        Sha256::Hash::kSize);
                     // Remaining time is not included in the telemetry
                 }
             }
         }
         // End of Nat64Mapping section.
 #endif // OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_DHCP6_PD
+        // Start of Dhcp6PdState section.
+        {
+            wpanBorderRouter->set_dhcp6_pd_state(
+                Dhcp6PdStateFromOtDhcp6PdState(otBorderRoutingDhcp6PdGetState(mInstance)));
+        }
+        // End of Dhcp6PdState section.
+
+        // Start of Hashed PD prefix
+        {
```

Comment:
In the following block, please group all the variable declarations and put them at the beginning of the block.